subprojects/introspect/source/jcli/introspect/data.d

```diff
@@ -130,11 +113,88 @@ template getArgumentFieldSymbol(TCommand, alias argumentInfoOrFieldPath)
     mixin("alias getArgumentFieldSymbol = TCommand." ~ fieldPathOf!argumentInfoOrFieldPath ~ ";");
 }
 
+template getCommandUDAs(CommandType)
+{
+    alias commandUDAs = AliasSeq!(
+        getUDAs!(CommandType, Command),
+        getUDAs!(CommandType, CommandDefault));
+
+    static if (commandUDAs.length > 0)
+    {
+        alias getCommandUDAs = commandUDAs;
+    }
+    else
+    {
+        enum isValue(alias stringUDA) = is(typeof(stringUDA) : string);
+        alias stringUDAs = Filter!(isValue,
+            getUDAs!(CommandType, string));
+
+        static if (stringUDAs.length > 0)
+            alias getCommandUDAs = AliasSeq!(stringUDAs[0]);
+        else
+            alias getCommandUDAs = AliasSeq!();
+    }
+}
+
+
+// NOTE:
+// It's not clear at all what a good design fot this is.
+// I only feel esoteric OOP vibes when deciding whether this should be in a struct,
+// template, whether it should be a getter, and enum, immutable, or an alias.
+// It feels to me like it's only philosophical at this point.
+// So I'm just going to do a single thing and call it a day.
+// Whatever I do feels wrong tho.
```

Comment:
A template is fine. Going with a `struct` makes it a bit easier to populate from CTFE functions though, so I'd say it depends on your needs.